worlds/sc2/client_gui.py

```diff
@@ -72,6 +73,21 @@ class MissionLayout(GridLayout):
 class MissionCategory(GridLayout):
     pass
 
+
+_has_forced_save = False
+def force_settings_save_on_close() -> None:
+    """
+    Settings has an existing auto-save feature, but it only triggers if a new key was introduced.
+    Force it to mark things as changed by introducing a new key and then cleaning up.
+    """
+    global _has_forced_save
+    if _has_forced_save:
+        return
+    SC2World.settings.update({'invalid_attribute': True})
+    del SC2World.settings.invalid_attribute
+    _has_forced_save = True
```

Comment:
I'm not proud of this function, but it works and seems to be the simplest way to force a save of the options. I don't know why there isn't an existing function call to do that, but I don't want to add it in this PR as it's a core change.